Source/WebCore/dom/ElementInternals.cpp

```diff
@@ -50,6 +52,88 @@ ShadowRoot* ElementInternals::shadowRoot() const
     return shadowRoot;
 }
 
+ExceptionOr<RefPtr<HTMLFormElement>> ElementInternals::form() const
+{
+    RefPtr element = m_element.get();
```

Comment:
Can we just use dynamicDowncast with these like this?
`m_element ? dynamicDowncast<HTMLMaybeFormAssociatedCustomElement>(*m_element) : nullptr`

Reply:
Extracted `elementAsFormAssociatedCustom()` with `dynamicDowncast` in 09aa647.

Reply:
Yet we can't just query `form()` from `HTMLMaybeFormAssociatedCustomElement` without checking its custom element state / form-associatedness.